src/SBInterpolatedImage.cpp

```diff
@@ -426,17 +433,101 @@ namespace galsim {
+        if (row_len <= 6) {
+            for (;it != end;){
+                oss.width(15);
+                oss << *it++;
+                if (it != end) oss << ",";
+            }
+        } else {
+            for (int i=0; i<3; i++){
+                oss.width(15);
+                oss << *it++;
+                oss << ",";
+            }
+            oss << " ...,";
+            it = end-3;
+            for (int i=0; i<3; i++){
+                oss.width(15);
+                oss << *it++;
+                if (it != end) oss << ",";
+            }
+        }
+        oss << "]";
+        return oss.str();
+    }
+
+    std::string image_string(ConstImageView<double> &im) {
+        Bounds<int> _bds = im.getBounds();
+        const int ymin = _bds.getYMin();
+        const int ymax = _bds.getYMax();
+
+        const int col_len = ymax - ymin;
+        std::ostringstream oss;
+        if (col_len <= 6) {
+            for (int y=ymin; y<=ymax; y++) {
+                if (y == ymin) oss << "array([";
+                else oss << "       ";
+                oss << row_string(im, y);
+                if (y != ymax) oss << "," << std::endl;
+            }
+        } else {
+            for (int y=ymin; y<ymin+3; y++) {
+                if (y == ymin) oss << "array([";
+                else oss << "       ";
+                oss << row_string(im, y) << "," << std::endl;
+            }
+            oss << "       ...," << std::endl;
+            for (int y=ymax-2; y<=ymax; y++) {
+                oss << "       ";
+                oss << row_string(im, y);
+                if (y != ymax) oss << "," << std::endl;
+            }
+        }
+        oss << "], dtype=float)";
+        return oss.str();
+    }
+
+    // Note you can trigger this function from python with an InterpolatedImage object in hand via:
+    // >>> repr(ii.SBProfile)
     std::string SBInterpolatedImage::SBInterpolatedImageImpl::repr() const
     {
         std::ostringstream oss(" ");
-        oss.precision(std::numeric_limits<double>::digits10 + 4);
         oss << "galsim._galsim.SBInterpolatedImage(";
+        oss << "galsim._galsim.ConstImageViewD(" << std::endl;
+        ConstImageView<double> im = getImage();
+        oss << image_string(im) << ", ";
 
+        boost::shared_ptr<Interpolant> xinterp = getXInterp();
+        boost::shared_ptr<Interpolant> kinterp = getKInterp();
+        oss << "galsim.Interpolant('"<<xinterp->makeStr()<<"', "<<xinterp->getTolerance()<<"), ";
+        oss << "galsim.Interpolant('"<<kinterp->makeStr()<<"', "<<kinterp->getTolerance()<<"), ";
+
+        oss << "1., "<<stepK()<<", "<<maxK()<<", galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    // Alternative to repr that's more annoying for a human to read, but can be used to *exactly*
+    // reconstruct the SBInterpolatedImage object when pickling.
+    std::string SBInterpolatedImage::SBInterpolatedImageImpl::serialize() const
```

Comment:
Maybe this is why you overrode the Add, Convolve, etc. reprs.  I'm not sure that we need the reprs for SBInterpolatedImage to be particularly small.  It's extremely rare for that to ever get triggered by someone.  Certainly not by accident.  We do our best to keep the C++ classes well hidden from the typical user.  So I think I'd prefer to just have `serialize` for all of these and have `SBProfile` use that for the repr.